docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+
+The StdoutExporter will also be used as a debugging tool for the Metrics API/SDK and Tracing API/SDK which are currently work in progress projects. This exporter will allow contributors to easily diagnose problems when working on the project.
+
+## Push vs Pull Exporter
+
+There are two different versions of exporters: Push and Pull. A Push Exporter pushes the data outwards towards a system, in the case of the StdoutExporter it prints data into Stdout. A Pull Exporter exposes data to some endpoint for another system to grab the data. 
+
+The StdoutExporter will only be implementing a Push Exporter framework.
+
+## Design Tenets
+
+* Reliability
+    * The Exporter should be reliable; data exported should always be accounted for. The data will either all be successfully exported to the destination server, or in the case of failure, the data is dropped. `Export` will always return failure or success to notify the user of the result.
+    * Thread Safety
+        * Export() should never be called concurrently. This is handled by the controller in the StdoutMetricsExporter.
+* Scalability
+    * The Exporter must be able to operate on sizeable systems with predictable overhead growth.  A key requirement of this is that the library does not consume unbounded memory resource. 
+* Security
+    * While authorization is required in protocol-specific exporters for third-party platforms, the StdoutExporter has no authentication since there is no third-party to communicate with.
+
+## SpanExporter
+
+`Span Exporter` defines the interface that protocol-specific exporters must implement so that they can be plugged into OpenTelemetry SDK and support sending of telemetry data.
+
+The goal of the interface is to minimize burden of implementation for protocol-dependent telemetry exporters. The protocol exporter is expected to be primarily a simple telemetry data encoder and transmitter.
+
+The SpanExporter is called through the SpanProcessor, which passes finished spans to the configured SpanExporter, as soon as they are finished. The SpanProcessor also shutdown the exporter by the Shutdown function within the SpanProcessor.
+
+![SDK Data Path](./images/SpanDataPath.png)
+
+The specification states: exporter must support two functions: Export and Shutdown.
+
+
+### `Export(span of recordables)`
+
+Exports a batch of telemetry data. Protocol exporters that will implement this function are typically expected to serialize and transmit the data to the destination.
+
+Export() will never be called concurrently for the same exporter instance. Export() can be called again only after the current call returns.
```

Comment:
This doesn't seem to be aligned with the spec.

Reply:
This wording is from the spec, see - https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#span-exporter

Am I looking at the wrong spec?

Reply:
You're looking at the right spec.

Let me clarify with the specification SIG next week.

Reply:
I brought up this topic during the 7/14/20 spec SIG meeting, I will need to submit a PR to change the spec.
For the OpenTelemetry C++ project, I think we need to assume that `Exporter()` can be called simultaneously.

Reply:
What changes need to be made to accommodate this change?

Reply:
Stdout Exporter could be called simultaneously, either 1) we make it clear that we don't do anything about, or 2) we need to introduce synchronization in the console exporter.
I would go with 1) as synchronization should be done at a lower level (e.g. on the console object, rather than the exporter).

Reply:
Updated and moved to design tenets